api/resources_portal/views/import_materials.py

```diff
@@ -0,0 +1,88 @@
+from django.forms.models import model_to_dict
+from django.http import JsonResponse
+from rest_framework import viewsets
+from rest_framework.permissions import IsAuthenticated
+
+from resources_portal.models import Grant, Material, Organization
+from resources_portal.views.import_sra_utils import ENA_URL_TEMPLATE, gather_all_metadata
+
+
+def import_sra(run_accession, organization, grant, user):
+    """
+    This function returns a Response object containing the json representation of the newly-created material object
+    made using SRA data.
+    """
+    metadata = gather_all_metadata(run_accession)
+
+    # note for the PR: currently, I use "library-construction-protocol" to populate the description. This is a technical description of the steps taken to process the material.
+    # Should I use the study abstract to populate it instead?
+
+    # One more question for PR. I was not able to find a way to retrieve "num_samples", "technology", "pre_print_doi", or "pre_print_title"
+    # from the SRA api. Are these neccessary to retrieve automatically? If so, I may need to pair with someone to find out out how to get those.
```

Comment:
For num_samples, take a look at https://github.com/AlexsLemonade/refinebio/blob/dev/foreman/data_refinery_foreman/surveyor/sra.py#L578. The number of runs is the number of samples for our purposes. If you do something similar to that and then do `len(accessions_to_run` that would be good for `num_samples`.

For technology, I see `'library_strategy': 'RNA-Seq'` in the metadata. @cgreene do you think this field will be what we want for technology?

I'm not actually sure how to get pre-print info. @cgreene is probably the person to ask about this as well.

Reply:
Cool. I added these two parameters, and left out pre-print info for now.